src/ImageCaptureCore/ICCameraDevice.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using Foundation;
+using ObjCRuntime;
+#nullable enable
+
+namespace ImageCaptureCore {
+	partial class ICCameraDevice {
+
+		public delegate void DidReadDataDelegate (NSData data, ICCameraFile file, NSError error);
```

Comment:
The documentation isn't clear, but I believe all these parameters should be nullable:

1. `error` will be null if the request was successful
2. `data` and `file` will likely be null if the request failed

```suggestion
		public delegate void DidReadDataDelegate (NSData? data, ICCameraFile? file, NSError? error);
```